src/utils/chunk.c

```diff
@@ -31,9 +32,15 @@
 #include <string.h>
 
 #define NN_CHUNK_TAG 0xdeadcafe
+#define NN_CHUNK_TAG_PTR 0xdeadd00d
```

Comment:
I am trying to reuse as much as possible from the original chunk structure, with the smallest memory impact possible.

In order to differentiate a classic chunk vs a pointer chunk I could have used two approaches:
- Introduce a `type` field in the `nn_chunk` structure (adds an extra byte in memory)
- Use a different tag (needs an extra tag check, but requires fewer tests in order to get the underlying pointer)
I chose the second.